### PR TITLE
Use Kokkos_ENABLE_PTHREAD=ON for one of the Travis CI builds

### PR DESCRIPTION
**Pull request by masterleinad**
Looking at #2857, it seems to be useful to have at least one build in the CI using `Kokkos_ENABLE_PTHREAD=ON`.

### CODE REVIEW

**Comment by masterleinad** [comment]
Travis failed as expected: https://travis-ci.org/github/kokkos/kokkos/jobs/661247783?utm_medium=notification&utm_source=github_status

File: .travis.yml [hunk 1]
```diff
@@ -31,8 +31,8 @@ branches:
 
 env:
   - 
-  - BACKEND="OPENMP"
-#  - BACKEND="PTHREAD"
+#  - BACKEND="OPENMP"
+  - BACKEND="PTHREAD"
```
Comment (dalg24) [inline .travis.yml]: ???
Did you mean to disable all the OpenMP builds?

Comment (masterleinad) [reply .travis.yml]: No, we still have some of them testing `OpenMP` (the ones that also do the coverage).

Comment (dalg24) [reply .travis.yml]: No AppleClang OpenMP for instance.  I guess my question is did you guys carefully look at what this change means

Comment (masterleinad) [reply .travis.yml]: But `AppleClang` with `OpenMP` is disabled anyway: https://github.com/kokkos/kokkos/blob/b5b7660af56d883ffc8892fa8a253ee257f38e26/.travis.yml#L43-L57

For me, this pull request still is an improvement.

**Pull request merged**